conans/test/functional/command/build_test.py

```diff
@@ -309,3 +309,29 @@ def build(self):
         client.save({CONANFILE: conanfile}, clean_first=True)
         client.run("install . --build missing")
         client.run("build .")
+
+    def build_full_reference_test(self):
+        client = TestClient()
+        conanfile = """
+from conans import ConanFile, CMake
+
+class FooConan(ConanFile):
+    name = "foo"
+    version = "1.0"
+"""
+        client.save({CONANFILE: conanfile})
+        client.run("create . user/stable")
+
+        conanfile = """
+from conans import ConanFile
+
+class BarConan(ConanFile):
+    name = "bar"
+    version = "1.0"
+    requires = "foo/1.0@user/stable"
+"""
+        client.save({CONANFILE: conanfile}, clean_first=True)
+        client.run("create --build foo/1.0@user/stable --build bar/1.0@user/testing . user/testing")
+        self.assertIn("foo/1.0@user/stable: WARN: Forced build from source", client.out)
+        self.assertIn("bar/1.0@user/testing: WARN: Forced build from source", client.out)
+        self.assertNotIn("ERROR", client.out)
```

Comment:
Not needed. If the client fails it will raise unless `assert_error` parameter is used.
I would add another create only with `--build foo/1.0@user/stable` to check that only foo is built.

Reply:
done